rinja_parser/src/lib.rs

```diff
@@ -635,14 +635,17 @@ impl<'a> State<'a> {
         }
     }
 
-    fn nest<'b>(&self, i: &'b str) -> ParseResult<'b, ()> {
+    fn nest<'b, T, F: FnMut(&'b str) -> ParseResult<'b, T>>(
+        &self,
+        i: &'b str,
+        mut callback: F,
+    ) -> ParseResult<'b, T> {
+        let prev_level = self.level.get();
         let (_, level) = self.level.get().nest(i)?;
```

Comment:
You could directly call `prev_level.nest(i)`.

Reply:
Indeed!